src/thread_manager.ts

```diff
@@ -56,279 +61,204 @@ export class ThreadManager<SCG extends ExtendableGenerics = DefaultGenerics> {
-        },
-      ),
-    );
-
-    const handleThreadsChange: Handler<readonly [Thread<SCG>[]]> = ([newThreads], previouslySelectedValue) => {
-      // create new threadIdIndexMap
-      const newThreadIdIndexMap = newThreads.reduce<ThreadManagerState['threadIdIndexMap']>((map, thread, index) => {
-        map[thread.id] ??= index;
-        return map;
-      }, {});
-
-      //  handle individual thread subscriptions
-      if (previouslySelectedValue) {
-        const [previousThreads] = previouslySelectedValue;
-        previousThreads.forEach((t) => {
-          // thread with registered handlers has been removed or its signature changed (new instance)
-          // deregister and let gc do its thing
-          if (typeof newThreadIdIndexMap[t.id] === 'undefined' || newThreads[newThreadIdIndexMap[t.id]] !== t) {
-            t.unregisterSubscriptions();
-          }
-        });
+      if (isSeen) {
+        this.state.partialNext({ isThreadOrderStale: true });
+      } else if (!unseenThreadIds.includes(parentId)) {
+        this.state.partialNext({ unseenThreadIds: unseenThreadIds.concat(parentId) });
       }
-      newThreads.forEach((t) => t.registerSubscriptions());
-
-      // publish new threadIdIndexMap
-      this.state.next((current) => ({ ...current, threadIdIndexMap: newThreadIdIndexMap }));
-    };
-
-    this.unsubscribeFunctions.add(
-      // re-generate map each time the threads array changes
-      this.state.subscribeWithSelector((nextValue) => [nextValue.threads] as const, handleThreadsChange),
-    );
-
-    // TODO: handle parent message hard-deleted (extend state with \w hardDeletedThreadIds?)
-
-    const handleNewReply = (event: Event<SCG>) => {
-      if (!event.message || !event.message.parent_id) return;
-      const parentId = event.message.parent_id;
+    }).unsubscribe;
 
-      const {
-        threadIdIndexMap,
-        nextCursor,
-        threads,
-        unseenThreadIds,
-        existingReorderedThreadIds,
-        active,
-      } = this.state.getLatestValue();
-
-      // prevents from handling replies until the threads have been loaded
-      // (does not fill information for "unread threads" banner to appear)
-      if (!threads.length && nextCursor !== null) return;
-
-      const existsLocally = typeof threadIdIndexMap[parentId] !== 'undefined';
-
-      // only register these changes during the time the thread manager is inactive
-      if (existsLocally && !existingReorderedThreadIds.includes(parentId) && !active) {
-        return this.state.next((current) => ({
+  private subscribeRecoverAfterConnectionDrop = () => {
+    const unsubscribeConnectionDropped = this.client.on('connection.changed', (event) => {
+      if (event.online === false) {
+        this.state.next((current) => ({
           ...current,
-          existingReorderedThreadIds: current.existingReorderedThreadIds.concat(parentId),
+          lastConnectionDropAt: current.lastConnectionDropAt ?? new Date(),
         }));
```

Comment:
We're re-creating state here even if the `lastConnectionDropAt` has possibly been already set.

Reply:
Fixed in https://github.com/GetStream/stream-chat-js/pull/1348/commits/caf81d8f9f3052e25adf772e56e83a7788767f63.